### PR TITLE
Add SUMA channels

### PR DESCRIPTION
**Pull request by jesusbv**
- If product is any if of the SUMA products allow access to those channels

## Description

Allow access to SUMa channels

## Change Type

*Please select the correct option.*

- [ ] **Bug Fix** (a non-breaking change which fixes an issue)
- [X] **New Feature** (a non-breaking change which adds new functionality)
- [ ] **Documentation Update** (a change which only updates documentation)

## Checklist

*Please check off each item if the requirement is met.*

- [X] I have verified that my code follows RMT's coding standards with `rubocop`.
- [X] I have reviewed my own code and believe that it's ready for an external review.
- [X] I have provided comments for any hard-to-understand code.
- [ ] I have documented the `MANUAL.md` file with any changes to the user experience.
- [X] RMT's test coverage remains at 100%.
- [ ] If my changes are non-trivial, I have added a changelog entry to notify users at `package/obs/rmt-server.changes`.

## Other Notes

Please use this space to provide notes or thoughts to the team, such as tips on how to review/demo your changes.

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/307983/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 23b6c324465360df9ef28cc5a4e35d104603837a into 314791ad30771a91a2ebb12e96a8461e7f364bd1)

File: engines/strict_authentication/app/controllers/strict_authentication/authentication_controller.rb [hunk 1]
```diff
@@ -35,7 +35,16 @@ def all_allowed_paths
       # so they can check if a customer has access to other products and show those
       # to them or verify paths
       all_product_versions = @system.products.map { |p| Product.where(identifier: p.identifier, arch: p.arch) }.flatten
-      all_product_versions.map { |prod| prod.repositories.pluck(:local_path) }.flatten
+      all_product_versions = all_product_versions.map { |prod| prod.repositories.pluck(:local_path) }.flatten
```
Comment (digitaltom) [inline engines/strict_authentication/app/controllers/strict_authentication/authentication_controller.rb:38]:
```suggestion
      allowed_paths = all_product_versions.map { |prod| prod.repositories.pluck(:local_path) }.flatten
```

**Comment by digitaltom** [comment]
Can you provide a real-world example of which kind of SUMA should be able to access which additional repositories? 
I understood that the SUMA server should be able to access the SUMA client tools repositories?

**Comment by digitaltom** [comment]
Was it intentional to delete the test contexts for "'when requested path is version 12.1'", 'when requested path is version 12', 'when accessing SLES12SP1 repos and SLES 11 is not activated'?

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/308437/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 23b6c324465360df9ef28cc5a4e35d104603837a into dc30add40c117b7545c8bb992175cefbce5ea0fe)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/309259/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 835151441e8a0f3f1aa467dff535319af51b4cee into dc30add40c117b7545c8bb992175cefbce5ea0fe)

**Comment by rjschwei** [comment]
> Can you provide a real-world example of which kind of SUMA should be able to access which additional repositories? I understood that the SUMA server should be able to access the SUMA client tools repositories?

SUMa server also needs access to the proxy channel

File: engines/strict_authentication/app/controllers/strict_authentication/authentication_controller.rb [hunk 2]
```diff
@@ -35,7 +35,16 @@ def all_allowed_paths
       # so they can check if a customer has access to other products and show those
       # to them or verify paths
       all_product_versions = @system.products.map { |p| Product.where(identifier: p.identifier, arch: p.arch) }.flatten
-      all_product_versions.map { |prod| prod.repositories.pluck(:local_path) }.flatten
+      allowed_paths = all_product_versions.map { |prod| prod.repositories.pluck(:local_path) }.flatten
+      manager_prod = @system.products.any? { |p| p.identifier.include?('manager') }
```
Comment (rjschwei) [inline engines/strict_authentication/app/controllers/strict_authentication/authentication_controller.rb:39]: we should restrict this to SUMa Server. A proxy system woud gets it's cache from SUMa server and itself does not need access. If we just look for `manager` we enable a sneak through pass when someone has a proxy server running.

**Review by rjschwei — CHANGES_REQUESTED** [review]
filter on server not just manager to avoid backdoor through proxy instances

File: engines/strict_authentication/app/controllers/strict_authentication/authentication_controller.rb (hunk 2 above)
Comment (digitaltom) [reply engines/strict_authentication/app/controllers/strict_authentication/authentication_controller.rb:39]: That would be 
```
manager_prod = @system.products.any? { |p| p.identifier.downcase.include?('manager-server') }
```
then.

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/309426/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 9106104f3fdb79bc5edeb39a299e6bb0ea9c8f3a into dc30add40c117b7545c8bb992175cefbce5ea0fe)

**Review by rjschwei — APPROVED** [review]

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/310769/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 9106104f3fdb79bc5edeb39a299e6bb0ea9c8f3a into 40bb4866a081f45874c9e7a40caa838c96c5d19f)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/311625/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged 9106104f3fdb79bc5edeb39a299e6bb0ea9c8f3a into 2e2a9361ec65f59f4cec75bc493b6fc215e26ce9)

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
https://ci.suse.de/job/scc-RMT-integration-tests/312099/console
If the given link has expired,you can force a Prophet rerun by just deleting this comment. (Merged d0e17c60a0e5ce90c7399a3fd4edf8bcd76f85e2 into 2e2a9361ec65f59f4cec75bc493b6fc215e26ce9)

**Review by digitaltom — APPROVED** [review]

**Pull request merged**